flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -953,7 +953,17 @@ public void releaseOrPromotePartitions(
         final Task task = taskSlotTable.getTask(executionAttemptID);
 
         if (task != null) {
-            task.triggerCheckpointBarrier(checkpointId, checkpointTimestamp, checkpointOptions);
+            CompletableFuture<Acknowledge> resultFuture = new CompletableFuture<>();
+            task.triggerCheckpointBarrier(checkpointId, checkpointTimestamp, checkpointOptions)
+                    .thenApply(
+                            triggerResult ->
+                                    triggerResult
+                                            ? resultFuture.complete(Acknowledge.get())
+                                            : resultFuture.completeExceptionally(
+                                                    new CheckpointException(
+                                                            "Task is not running?",
+                                                            CheckpointFailureReason
+                                                                    .TRIGGER_CHECKPOINT_FAILURE)));
 
             return CompletableFuture.completedFuture(Acknowledge.get());
```

Comment:
🤦 note that in this version `CompletableFuture.completedFuture(Acknowledge.get())` is returned regardless if the `resultFuture` succeeded or failed...

Reply:
Sorry for also missing this issue...